app/services/bulk_update/recommendations_uploads/validate_csv_row.rb

```diff
@@ -14,38 +16,142 @@ def valid?
+        first_names
+        last_names
+        lead_school
+        qts_or_eyts
+        route
+        phase
+        age_range
+        subject
       end
 
-      def trn
+      def trn_format
+        return if row.trn.blank?
         return if row.trn =~ /^\d{7}$/
 
-        @messages << "TRN must be 7 characters long and contain only numbers"
+        @messages << error_message(:trn_format)
       end
 
-      def hesa_id
+      def hesa_id_format
         return if row.hesa_id.nil?
-        return if row.hesa_id =~ /^\d{17}$/
+        return if row.hesa_id =~ /^[0-9]{13}([0-9]{4})?$/
 
-        @messages << "HESA ID must be 17 characters long and contain only numbers"
+        @messages << error_message(:hesa_id_format)
       end
 
       def standards_met_at
-        date = row.standards_met_at&.to_date
-        if date && date > Time.zone.today
-          @messages << "Award date must not be in the future"
-        elsif date && date < 12.months.ago
-          @messages << "Award date cannot be from more that 12 months ago #{12.months.ago.to_date.iso8601}"
+        case row.standards_met_at
+        when /^\d{1,2}[\/-]\d{1,2}[\/-]\d{4}$/ # dd/mm/yyyy or dd-mm-yyyy or d/m/yyyy etc etc
+          date = row.standards_met_at.to_date
+          today = Time.zone.today.to_date
+          ago_12_months = 12.months.ago.to_date
+
+          @messages << error_message(:award_date_future, date: gds_date(date)) if date > today
+          @messages << error_message(:award_date_past, date: gds_date(ago_12_months)) if date < ago_12_months
+          @messages << error_message(:date_standards_met, date: gds_date(trainee.itt_start_date.to_date)) if trainee&.itt_start_date && date < trainee.itt_start_date.to_date
+        else
+          @messages << error_message(:date_parse)
+        end
+      end
+
+      def gds_date(date)
+        date.strftime(Date::DATE_FORMATS[:govuk])
+      end
+
+      def column_exists?(column_name)
+        csv.headers.include?(column_name.downcase)
+      end
+
+      def trn
+        return unless column_exists?(Reports::BulkRecommendReport::TRN)
+
+        @messages << error_message(:trn) if trainee.trn != row.trn
+      end
+
+      def hesa_id
+        return unless column_exists?(Reports::BulkRecommendReport::HESA_ID)
+
+        @messages << error_message(:hesa_id) if trainee.hesa_id != row.hesa_id
+      end
+
+      def provider_trainee_id
+        return unless column_exists?(Reports::BulkRecommendReport::TRAINEE_ID)
+
+        @messages << error_message(:provider_trainee_id) if trainee.provider_trainee_id != row.provider_trainee_id
+      end
+
+      def first_names
+        return unless column_exists?(Reports::BulkRecommendReport::FIRST_NAME)
+
+        # UnicodeUtils is used to remove accented characters for a simpler/more reliable comparison
+        if I18n.transliterate(trainee.first_names.downcase, replacement: "") != I18n.transliterate(row.first_names.downcase, replacement: "")
```

Comment:
I might pull this funky `transliterate` code into a method we can call `remove_accented_characters(string)` or something as we're using it 4 times.